src/Backend/Modules/MediaLibrary/Ajax/MediaItemUpload.php

```diff
@@ -56,87 +104,45 @@ public function execute(): void
     {
         parent::execute();
 
-        // Include the upload handler class
-        $uploader = new UploadHandler($this->getRequest(), $this->get('media_library.manager.file'));
-        // Specify the list of valid extensions, ex. array("jpeg", "xml", "bmp")
-        $uploader->allowedExtensions = $this->get('media_library.manager.extension')->getAll();
-        $uploader->allowedMimeTypes = $this->get('media_library.manager.mime_type')->getAll();
-        // Specify max file size in bytes.
-        $uploader->sizeLimit = null;
-        // Specify the input name set in the javascript.
-        $uploader->inputName = 'qqfile'; // matches Fine Uploader's default inputName value by default
-        // If you want to use the chunking/resume feature, specify the folder to temporarily save parts.
-        $uploader->chunksFolder = 'chunks';
-
-        $this->response = new Response();
-
-        $method = $this->getRequestMethod();
-
-        /*
-         * handle the preflighted OPTIONS request. Needed for CORS operation.
-         */
-        if ($method === 'OPTIONS') {
-            $this->handlePreflight();
-            throw new RedirectException('exit', $this->response);
-        }
-
-        if ($method !== 'POST') {
-            $this->response->setStatusCode(Response::HTTP_METHOD_NOT_ALLOWED);
-            throw new RedirectException('exit', $this->response);
-        }
-
-        $this->handleCorsRequest();
-        $this->response->headers->set('Content-Type', 'text/plain');
-
-        // Assumes you have a chunking.success.endpoint set to point here with a query parameter of "done".
-        // For example: /myserver/handlers/endpoint.php?done
-        if ($this->getRequest()->query->get('done') !== null) {
-            $this->response->setContent($uploader->combineChunks('files'));
-            throw new RedirectException('exit', $this->response);
-        }
-
-        // Define upload dir
-        $uploadDir = $this->get('media_library.storage.local')->getUploadRootDir() . '/' . $this->get(
-            'media_library.manager.file'
-        )->getNextShardingFolder();
+        $this->request = $this->getRequest();
+        $this->response = $this->createResponse();
+        $this->uploadHandler = $this->createUploader();
+        $this->uploadDirectory = $this->getUploadDirectory();
+        $this->combineChunksAfterReceivingThemAll();
 
-        // Generate folder if not exists
-        $this->get('media_library.manager.file')->createFolder($uploadDir);
-
-        // Call handleUpload() with the name of the folder, relative to PHP's getcwd()
-        $result = $uploader->handleUpload($uploadDir);
+        $result = $this->uploadHandler->handleUpload($this->uploadDirectory);
 
-        if (array_key_exists('error', $result)) {
+        if (array_key_exists('error', $result)
```

Comment:
Personal preference: would extract if-logic to separate method that returns bool, for readability

Reply:
9303ba4e592f14461862c2aeda4aacd3dce78595

Reply:
awesome